### DISCUSSION TITLE
Skyblock Levels missing information

### DISCUSSION THREAD

**Discussion by MattTheCuber** [HypixelDev/PublicAPI#565] [Q&A] [answered]
Hello, this question is directed at the skyblock developers.

We are trying to make a Skyblock Level calculator and obviously have run into a few issues.

1. **Museum** is obvious and not the topic of this post.
2. **Quests** are difficult because some give no XP, others 5 XP, and maybe some give 10 XP? I would be amazing if there was some way to see this in game (maybe in the quest lore, or a different color for each type?).
3. Bank Upgrades
4. Now, the real question for this post is the **exponential powder formulas**. According to the game you earn an exponentially decreasing amount of Skyblock XP between 350k and 12.5m/20m. We have tried calculating these formulas to save asking the developers for it, but have realized that we can only obtain approximate formulas. It would be amazing if the developers could either display these formulas in game or share it in this post so that we can make our calculation a little more accurate.

![image](https://user-images.githubusercontent.com/32849887/194754709-0af11e6f-eab8-4430-88e5-357b6356b9a7.png)

**Comment by hannibal002** [comment]
Please add skyblock experience as a an single value to the API. Then I can calculate the skyblock level without fancy math thingy. Thanks. Would make me and I guess everyone else happy.

**Comment by LandonDev** [comment, marked as answer]
Hey guys, SkyBlock dev here. Levels is going to be added to the API soon, but in the meantime, here is the formula for the overflow powder calculations. Keep in mind this is just for overflow, not the per powder until 350k:

multiplier * (Math.sqrt(1 + 8 * (Math.sqrt((1758267 / CAP) * left + 9))) - 3)

_Where multiplier is 3.75 for Mithril and 4.25 for Gemstone and CAP is 12,500,000 for Mithril and 20,000,000 for Gemstone_
_“Left” is the amount of powder left after subtracting 350,000_

**Reply by qualya** [reply]
This formula doesn't seem to be working correctly
When putting in the values for having max Mithril powder, the overflow is ~372.27 XP
When putting in the values for max Gemstone powder, the overflow is ~423.07 XP

From the description, it says that it is a linear increase until 350k, until this function is used for the rest
Since we know that the max amount of XP is 1495, and the XP you get from HOTM tiers is 545 at max, the total XP for both of the linear equations should be 1495-372.27-423.07-545, which would be 154.66 XP

However, for Mithril it's supposed to be 1 XP/2400 powder up to 350,000, and 1 XP/2500 powder up to 350,000 for Gemstone
This would mean that the total XP for both linear equations should be 350,000/2400 + 350,000/2,500 which is ~285.8333 XP

This is ~131.17 XP more than the 154.66 XP left over for the two linear equations
So, either this overflow equation is not right, or the description of one or both of the linear equations isn't true
Assuming the description is wrong, then that would mean an average of around 1XP/4526 powder up to 350k for each of Mithril and Gemstone instead of 1XP/2400 and 1XP/2500

**Reply by LandonDev** [reply]
There must be something wrong with your calculations. Remember, the input is after subtracting the cap, so max for Mithril would be 12,150,000. I will once again put the formula exactly how it appears in our calculation:

const = 1758267
left = powder - 350000
multiplier = 3.75 for mithril and 4.25 for gemstone
cap = 12,500,000 for mithril and 20,000,000 for gemstone

(multiplier * (Math.sqrt(1 + 8 * (Math.sqrt((cons / cap) * left + 9))) - 3))

**Reply by qualya** [reply]
I have checked for some people, and I don't think the overflow equation is the problem
It seems to work, but only if you use ~154.66 XP for the linear portion up to 350k for both mithril and gemstone, rather than ~285.8333 XP like I found in my last reply
So it seems that the descriptions of "Gain +1 XP per 2,400 powder until 350,000 powder." and "Gain +1 XP per 2,500 powder until 350,000 powder." are wrong or outdated or something
I'm not sure what the actual linear equations for these are then, but as I said in the last reply, it should be an average of gaining around +1 XP per 4,526  powder instead of 2,450
Either the description is wrong and the linear equation is something different, or there's some weird other reason that ~131.17 XP is not being added
Could you check to see what the two linear equations are?

**Reply by MattTheCuber** [reply]
FYI the 2,450 he refers to is the average of both types (2,400 for mithril and 2,500 for gemstone). So the average appears to actually be 4,526. Which would mean something like 4,500 for mithril and 4,550 for gemstone.

**Reply by MattTheCuber** [reply]
Ok, now I'm just super confused. Because this guy has HotM 4 which is supposed to be 215 XP, but his total HotM XP including powder is 172 XP.

IGN: DuckySoSkilled
![unknown](https://user-images.githubusercontent.com/32849887/195983713-70398d18-8ed1-4b8f-b81a-932076224ffd.png)

**Reply by tenios-sb** [reply]
So, I looked into this myself and from what I can tell, the given formula here is correct.

However, there seems to be a bug where your highest HOTM level does not grant any sb xp, so at hotm 6 you only gain the xp from lvl 1-5.
If this gets fixed, the actual max amount would be 1625 sb xp. 

Ignoring the highest hotm level, I get correct values from my calculator using the exact formula given here.
